File: src/main/java/io/github/tanguygab/conditionalactions/actions/types/messages/BroadcastAction.java
package io.github.tanguygab.conditionalactions.actions.types.messages;

import io.github.tanguygab.conditionalactions.actions.types.Action;
import org.bukkit.OfflinePlayer;

public class BroadcastAction extends Action {

    public BroadcastAction() {
        super("^(?i)(broadcast|bc):( )?");
    }

    @Override
    public String getSuggestion() {
        return "broadcast: <message>";
    }

    @Override
    public void execute(OfflinePlayer player, String match) {
        String message = parsePlaceholders(player, match, true);
        getPlugin().getServer().getOnlinePlayers().forEach(p->p.sendMessage(message));
    }
}

File: src/main/kotlin/io/github/tanguygab/conditionalactions/actions/types/messages/BroadcastAction.kt
package io.github.tanguygab.conditionalactions.actions.types.messages

import io.github.tanguygab.conditionalactions.actions.Action
import net.kyori.adventure.text.minimessage.MiniMessage
import org.bukkit.OfflinePlayer

class BroadcastAction : Action("^(?i)(broadcast|bc):( )?".toRegex()) {
    override fun getSuggestion() = "broadcast: <message>"

    override fun execute(player: OfflinePlayer?, match: String) {
        val message = parsePlaceholders(player, match)
        plugin.server.sendMessage(mm.deserialize(message))
    }
}
